File: backend/src/main/java/com/minutebook/model/Meeting.java
package com.minutebook.model;

import com.minutebook.model.enums.MeetingStatus;
import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "meetings")
public class Meeting {

    @Id
    @Column(length = 36, nullable = false)
    private String id;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String title;

    @Column(name = "audio_path", nullable = false, columnDefinition = "TEXT")
    private String audioPath;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private MeetingStatus status = MeetingStatus.QUEUED;

    @Column(name = "duration_seconds")
    private Integer durationSeconds;

    @Column(columnDefinition = "TEXT")
    private String summary;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "meeting", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("sequence ASC")
    private List<TranscriptSegment> transcriptSegments = new ArrayList<>();

    @OneToMany(mappedBy = "meeting", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<KeyDecision> keyDecisions = new ArrayList<>();

    @OneToMany(mappedBy = "meeting", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ActionItem> actionItems = new ArrayList<>();

    @OneToMany(mappedBy = "meeting", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("startTime ASC")
    private List<Chapter> chapters = new ArrayList<>();

    @PrePersist
    public void prePersist() {
        if (this.id == null) {
            this.id = UUID.randomUUID().toString();
        }
        if (this.createdAt == null) {
            this.createdAt = LocalDateTime.now();
        }
    }

    // --- Constructors ---
    public Meeting() {}

    public Meeting(String title, String audioPath) {
        this.title = title;
        this.audioPath = audioPath;
        this.status = MeetingStatus.QUEUED;
    }

    // --- Getters and Setters ---
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getAudioPath() { return audioPath; }
    public void setAudioPath(String audioPath) { this.audioPath = audioPath; }

    public MeetingStatus getStatus() { return status; }
    public void setStatus(MeetingStatus status) { this.status = status; }

    public Integer getDurationSeconds() { return durationSeconds; }
    public void setDurationSeconds(Integer durationSeconds) { this.durationSeconds = durationSeconds; }

    public String getSummary() { return summary; }
    public void setSummary(String summary) { this.summary = summary; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public List<TranscriptSegment> getTranscriptSegments() { return transcriptSegments; }
    public void setTranscriptSegments(List<TranscriptSegment> segments) { this.transcriptSegments = segments; }

    public List<KeyDecision> getKeyDecisions() { return keyDecisions; }
    public void setKeyDecisions(List<KeyDecision> decisions) { this.keyDecisions = decisions; }

    public List<ActionItem> getActionItems() { return actionItems; }
    public void setActionItems(List<ActionItem> items) { this.actionItems = items; }

    public List<Chapter> getChapters() { return chapters; }
    public void setChapters(List<Chapter> chapters) { this.chapters = chapters; }
}

File: frontend/src/types/meeting.ts
export type MeetingStatus = 'QUEUED' | 'TRANSCRIBING' | 'SUMMARIZING' | 'DONE' | 'FAILED';

export interface MeetingListItem {
  id: string;
  title: string;
  status: MeetingStatus;
  durationSeconds: number | null;
  createdAt: string;
}

export interface TranscriptSegment {
  id: string;
  speakerLabel: string | null;
  startTime: number;
  endTime: number;
  text: string;
  sequence: number;
}

export interface KeyDecision {
  id: string;
  decision: string;
  context: string | null;
  sourceTimestamp: number | null;
}

export interface ActionItem {
  id: string;
  task: string;
  owner: string | null;
  deadline: string | null;
  priority: 'high' | 'medium' | 'low';
  sourceTimestamp: number | null;
  isComplete: boolean;
}

export interface Chapter {
  id: string;
  title: string;
  startTime: number;
}

export interface MeetingDetail {
  id: string;
  title: string;
  status: MeetingStatus;
  durationSeconds: number | null;
  summary: string | null;
  createdAt: string;
  transcriptSegments: TranscriptSegment[];
  keyDecisions: KeyDecision[];
  actionItems: ActionItem[];
  chapters: Chapter[];
}

export interface StatusResponse {
  id: string;
  status: MeetingStatus;
}
